app/models/task_file.rb

```diff
@@ -46,6 +70,13 @@ def extract_text_data
 
   private
 
+  def attach_parent_blob
+    # The comparison of the blob ID is used as a safeguard to ensure updated files in a task contribution are not overwritten unintended.
+    return unless parent_blob_id.to_i == parent.attachment.blob.id
+
+    attachment.attach(parent.attachment.blob)
+  end
+
```

Comment:
As discussed: A valid test for this method would be a feature spec (soon called a system spec) with the following steps:

1. Access an existing task with an attached file that is not owned by the current user
2. Click on "Contribute"
3. Perform any change, e.g., for the title
4. Invalidate the task by removing the `language` (or do any other change)
5. Click "Save"
6. Expect that the new contribution was not saved and that errors where found. Further ensure that the attached file name is shown as part of a `div` with the classes `.btn.btn-light.border-right-0.disabled` (`app/views/tasks/_editor.html.slim:23`) and that re-uploading a new file is possible (`app/views/tasks/_editor.html.slim:24-25`).
7. Fix the previous validation error, e.g. by fixing the language
8. Hit "Save" again
9. Expect that the contribution was successfully saved. Expect that it features the new name and still has an attachment.

A second test explicitly checking that the method under test is not used for changed files would be:

1. Access an existing task with an attached file that is not owned by the current user
2. Click on "Contribute"
3. Perform any change, e.g., for the title
4. Replace the attachment of the existing file by selecting a new file for upload. [This post](https://til.hashrocket.com/posts/c790268652-attach-a-file-with-capybara) shows how to do that with Capybara.
5. Invalidate the task by removing the `language` (or do any other change)
6. Click "Save"
7. Expect that the new contribution was not saved and that errors where found. Further ensure that no file name is shown for the attachment and re-selecting the file is required (`app/views/tasks/_editor.html.slim:29-30`). 
8. Fix the previous validation error, e.g. by fixing the language and re-select the file to upload
9. Hit "Save" again
10. Expect that the contribution was successfully saved. Expect that it features the new name and still has the uploaded attachment.

Reply:
Awesome, thanks for the first feature spec! 

There is currently a failure on GitHub Actions, which is not related to your code. We are missing the required `libvips` library, which will be included with #1589: [proposed change for `ci.yml` including `libvips`](https://github.com/openHPI/codeharbor/pull/1589/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f).

Reply:
I extracted the required change with #1684, which we merged on master already. Rebasing will include this change, so that the test should pass.

Reply:
We added both tests and rebased the current branch onto the latest master, resolving the previous test failures. Hence, this conversation can be resolved.